File: linked_list/hasCycle/hasCycle.js
function hasCycle(head) {
    let hasCycle = false;

    if (head) {
        let slowPointer = head;
        let fastPointer = head;

        while (true) {
            if (fastPointer.next && fastPointer.next.next) {
                fastPointer = fastPointer.next.next;
                slowPointer = slowPointer.next;

                if (fastPointer === slowPointer) {
                    hasCycle = true;
                    break;
                }
            } else {
                break;
            }
        }
    }

    return hasCycle;
}

module.exports = hasCycle;

File: src/linked_list/hasCycle/hascycle.py
from ..singleLinkedList.singlelinkedlist import ListNode


class Solution(object):
    def hasCycle(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        hasCycle = False

        if head and head.next:
            slow_pointer = head
            fast_pointer = head

            while True:
                if fast_pointer.next and fast_pointer.next.next:
                    fast_pointer = fast_pointer.next.next
                    slow_pointer = slow_pointer.next

                    if fast_pointer == slow_pointer:
                        hasCycle = True
                        break
                else:
                    break

        return hasCycle
